### PR TITLE
Add translation for reminders

### PR DESCRIPTION
**Pull request by tsmr**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes

Add translation for reminders

### CODE REVIEW

**Comment by trasher** [comment]
There are cs issues ;)

File: inc/remindertranslation.class.php [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+
+if (!defined('GLPI_ROOT')) {
+   die("Sorry. You can't access this file directly");
+}
+
+
+/**
+ * ReminderTranslation Class
+ *
+ * @since version 9.2.2
```
Comment (trasher) [inline inc/remindertranslation.class.php]: 9.4 ;)

File: inc/remindertranslation.class.php [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ */
+
+/** @file
```
Comment (trasher) [inline inc/remindertranslation.class.php]: Useless

File: inc/remindertranslation.class.php [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+    * @param $item       item to translate
+    * @param $field      field to return (default 'name')
+    *
+    * @return the field translated if a translation is available, or the original field if not
+   **/
+   static function getTranslatedValue(Reminder $item, $field = "name") {
+      global $DB;
+
+      $obj   = new self;
+      $found = $obj->find("`reminders_id` = '".$item->getID().
```
Comment (trasher) [inline inc/remindertranslation.class.php]: Please avoid using methods using raw sql

File: inc/remindertranslation.class.php [hunk 4]
```diff
@@ -0,0 +1,331 @@
...
+
+   /**
+    * Is Reminder translation functionnality active
+    *
+    * @return true if active, false if not
+   **/
+   static function isReminderTranslationActive() {
+      global $CFG_GLPI;
+
+      //return $CFG_GLPI['translate_reminder'];
```
Comment (trasher) [inline inc/remindertranslation.class.php]: To drop

File: inc/remindertranslation.class.php [hunk 5]
```diff
@@ -0,0 +1,331 @@
...
+    *
+    * @param item
+    *
+    * @return array of already translated languages
+   **/
+   static function getAlreadyTranslatedForItem($item) {
+      global $DB;
+
+      $tab = [];
+      foreach ($DB->request(getTableForItemType(__CLASS__),
```
Comment (trasher) [inline inc/remindertranslation.class.php]: Please avoid raw sql

**Comment by trasher** [comment]
ping?

**Comment by amma35** [comment]
Is that good for you?

File: inc/reminder.class.php [hunk 6]
```diff
@@ -1141,19 +1159,19 @@ static function showListForCentral($personal = true) {
...
-               'users_id'  => $users_id,
-               [
-                  'OR'        => [
-                     'end'          => ['>=', $today],
-                     'is_planned'   => 0
-                  ]
-               ]
-            ], $visibility_criteria),
-            'ORDER'  => 'name'
+                                       'glpi_reminders.users_id'  => $users_id,
```
Comment (trasher) [inline inc/reminder.class.php]: ok to add table name, not ok for indent

File: inc/reminder.class.php [hunk 7]
```diff
@@ -103,7 +103,7 @@ function canUpdateItem() {
     * @since 0.85
     *
     * @see CommonDBTM::canPurgeItem()
-   **/
+    **/
```
Comment (trasher) [inline inc/reminder.class.php]: Such changes makes PRs difficult to run, please do not apply editor changes on a whole file ;)

File: inc/reminder.class.php [hunk 8]
```diff
@@ -939,7 +940,7 @@ static function populatePlanning($options = []) {
       if (count($nreadpub)
           && count($nreadpriv)) {
          $NASSIGN = ['OR' => [$nreadpub, $nreadpriv]];
-      } else if ($count(readpub)) {
+      } else if (count($nreadpub)) {
```
Comment (trasher) [inline inc/reminder.class.php]: it's been fixed already, just rebase on the 9.4/bugfixes branch

File: inc/reminder.class.php [hunk 9]
```diff
@@ -971,18 +972,27 @@ static function populatePlanning($options = []) {
 
       $table = self::getTable();
       $criteria = [
-         'SELECT DISTINCT' => "$table.*",
-         'FROM'            => $table,
-         'WHERE'           => $WHERE,
-         'ORDER'           => 'begin'
-      ] + $visibility_criteria;
+                     'SELECT DISTINCT' => "$table.*",
```
Comment (trasher) [inline inc/reminder.class.php]: Same for indent

File: inc/remindertranslation.class.php [hunk 10]
```diff
@@ -0,0 +1,314 @@
...
```
Comment (trasher) [inline inc/remindertranslation.class.php]: Do not call methods in loops

**Comment by trasher** [comment]
Seems to duplicate #5120

**Pull request closed without merging**